lib/siwapp_web/controllers/graphic_controller.ex

```diff
@@ -0,0 +1,57 @@
+defmodule SiwappWeb.GraphicController do
```

Comment:
This is not a controller, it could be named as Graphics or whatever but not controller.
Check the book to find a suitable place and name for a thing like this.

Reply:
I've renamed it as GraphicHelpers, but with an alias in the templates as Graphics. It's similar to what we do with Battleship. I didn't find it really clear in the book.